src/WooCommerce.php

```diff
@@ -342,18 +342,16 @@ public static function woocommerce_template_loop_product_thumbnail() {
 
       if (count($variation_ids)) {
         $placeholders = implode(',', array_fill(0, count($variation_ids), '%d'));
-        $attachment_ids = array_unique(
-          array_merge(
-            $attachment_ids,
-            $wpdb->get_col(
-              $wpdb->prepare(
-                "SELECT pm.meta_value AS attachment_id
-                FROM wp_posts p
-                INNER JOIN wp_postmeta pm ON pm.post_id = p.ID AND pm.meta_key = '_thumbnail_id'
-                WHERE p.ID IN ($placeholders)
-                ORDER BY p.menu_order ASC",
-                $variation_ids
-              )
+        $attachment_ids = array_merge(
+          $attachment_ids,
+          $wpdb->get_col(
+            $wpdb->prepare(
+              "SELECT DISTINCT pm.meta_value AS attachment_id
+              FROM wp_posts p
+              INNER JOIN wp_postmeta pm ON pm.post_id = p.ID AND pm.meta_key = '_thumbnail_id'
+              WHERE p.ID IN ($placeholders) AND pm.meta_value > 0
+              ORDER BY p.menu_order ASC",
+              $variation_ids
```

Comment:
```suggestion
              WHERE p.ID IN ($placeholders) AND pm.meta_value > 0 AND pm.meta_value <> %d
              ORDER BY p.menu_order ASC",
              $variation_ids, $attachment_ids[0]
```

Reply:
This is more difficult to read. I'll add back the array_unique if that's okay.

Reply:
@wrux suggested removing array unique and replacing with select distinct https://github.com/netzstrategen/wordpress-gallerya/pull/74#discussion_r618217776

Reply:
Ah I see now ok.

Reply:
Ok well in the interest of preferring everything to be done in the query I suppose this is fine. Will test this.

Reply:
@sun This did not test well. Adding back array_unique. Should I reverse the select distinct change then as well?